### PR TITLE
Migrate CircleCI builds from v1.0 to v2.0

### PR DESCRIPTION
**Pull request by mikesaelim**
CircleCI is sunsetting v1.0 build specifications at the end of August, so we should migrate over to v2.0.

https://circleci.com/blog/sunsetting-1-0/

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+version: 2
+jobs:
+  build:
+    docker:
+      - image: circleci/python:3.6.5
```
Comment (skomanduri) [inline .circleci/config.yml]: How did you pick this base image?

Comment (evelynting610) [reply .circleci/config.yml]: i don't see this tag on the taglist here: https://hub.docker.com/r/circleci/python/tags/
Let me know if I'm missing something.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run: docker build -t civisanalytics/civis-services-shiny .
+      - run: docker run --entrypoint=/bin/bash civisanalytics/civis-services-shiny -c "echo BUILDS OK"
+      - run: docker run --entrypoint=/bin/bash civisanalytics/civis-services-shiny -c "Rscript -e \"library(shiny); q(save = 'no')\""
```
Comment (evelynting610) [inline .circleci/config.yml]: is there a reason why we're not including a "test" as well as the "build"?

File: .circleci/config.yml (hunk 1 above)
Comment (mikesaelim) [reply .circleci/config.yml]: Yeah I was confused too, when I saw it used in other configs.  [But there's a note in the full description the main page](https://hub.docker.com/r/circleci/python/):

>Note: due to Docker Hub limitations, there may be more variants available than are displayed on the Tags page. For a complete list, along with Dockerfiles, see our circleci-public/circleci-dockerfiles repository.

Docker Hub... 🤦‍♂️

File: .circleci/config.yml (hunk 2 above)
Comment (mikesaelim) [reply .circleci/config.yml]: Ah, I guess I was thinking of this circleci job as doing the build and testing the build.  The test lines require the `docker build` statement to build the image... but unfortunately there doesn't seem to be an easy way to share images between different jobs.

https://discuss.circleci.com/t/docker-images-between-jobs-in-workflows/15383

I figured it's not worth it to do the extra cache stuff until we're doing more complicated stuff with our circle configs.

**Pull request merged**